regular_expressions/regular_expressions.md

```diff
@@ -0,0 +1,623 @@
+<!--
+
+author:   Joy Payton
+email:    paytonk@chop.edu
+version:  1.0.0
+current_version_description: Initial version
+module_type: standard
+docs_version: 1.0.0
+language: en
+narrator: UK English Female
+mode: Textbook
+
+title: Introduction to Regular Expressions
+
+comment:  Learn about pattern matching using regular expressions, or regex.
+
+long_description: Regular expressions, or regex, are a way to specify patterns (such as the pattern that defines a valid email address, medical record number, or credit card number).  Being able to use regex will make it easier for you to find important data.
+
+estimated_time_in_minutes: 60
+
+@pre_reqs
+Learners will be able to apply the skills learned in this module most rapidly if they have beginner-level experience in at least one query or programming language, such as SQL, R, Python, or Bash.  However, this module does not require coding experience and no computer code is written in this module.
+@end
+
+@learning_objectives  
+After completion of this module, learners will be able to:
+
+- Define an alphanumeric pattern in regex notation
+- Use capturing groups to extract data from a regular expression
+- Use anchors and boundaries in regular expressions
+- Test regular expressions in a browser
+@end
+
+@version_history 
+No previous versions.
+@end
+
+import: https://raw.githubusercontent.com/arcus/education_modules/main/_module_templates/macros.md
+-->
+
+# Regular Expressions
+
+@overview
+
+## What is a Regular Expression?
+
+The word *regular* (like *regulate*) comes from the Latin root for "rule": *regula*.  A **regular expression** is a specific way to **express** a **rule** for a pattern.  For example, one pattern rule might be for a phone number in the United States.  A phone number in the United States has:
+
+1. A three-digit area code, possibly enclosed by parentheses
+2. An optional separator, like a whitespace character (tab, space, etc.), hyphen, or a period
+3. A set of three digits
+4. An optional separator, like a whitespace character (tab, space, etc.), hyphen, or a period
+5. A set of four digits
+
+For example, these are all valid US phone numbers, and you might see them written various ways.
```

Comment:
Suggestions:
Would it make sense to have different numbers here? All in phone number styles but not all the same number?

Also, I'd suggest a more obviously fake number, or possibly a real number, but like the main switchboard for the hospital.

Also, my first reaction was that you should have at least a few examples starting with 1, (e.g. 1-212-123-4567) particularly as we are recruiting internationally.